brotli-sys/src/lib.rs

```diff
@@ -206,71 +201,256 @@ extern {
                                          symbols: *mut u16,
                                          num_symbols: u32) -> u32;
+}
+
+
+
+// ========== Encoder functionality ==========
+
+pub type BrotliEncoderMode = __enum_ty;
+pub type BrotliEncoderParameter = __enum_ty;
+pub type BrotliEncoderOperation = __enum_ty;
+pub type BrotliEncoderStreamState = __enum_ty;
+
+pub const RUST_MODE_GENERIC: BrotliEncoderMode = 0;
+pub const RUST_MODE_TEXT: BrotliEncoderMode = 1;
+pub const RUST_MODE_FONT: BrotliEncoderMode = 2;
+
+pub const RUST_PARAM_MODE: BrotliEncoderParameter = 0;
+pub const RUST_PARAM_QUALITY: BrotliEncoderParameter = 1;
+pub const RUST_PARAM_LGWIN: BrotliEncoderParameter = 2;
+pub const RUST_PARAM_LGBLOCK: BrotliEncoderParameter = 3;
+
+pub const RUST_OPERATION_PROCESS: BrotliEncoderOperation = 0;
+pub const RUST_OPERATION_FLUSH: BrotliEncoderOperation = 1;
+pub const RUST_OPERATION_FINISH: BrotliEncoderOperation = 2;
+
+pub const RUST_DEFAULT_QUALITY: u32 = 11;
+pub const RUST_DEFAULT_WINDOW: u32 = 22;
+
+pub type BrotliEncoderState = BrotliEncoderStateStruct;
+
+#[repr(C)]
+pub struct BrotliEncoderStateStruct {
+    pub mode: BrotliEncoderMode,
+    pub quality: c_int,
+    pub lgwin: c_int,
+    pub lgblock: c_int,
+    pub memory_manager_: MemoryManager,
+    pub hashers_: Hashers,
+    pub hash_type_: c_int,
+    pub input_pos_: u64,
+    pub ringbuffer_: RingBuffer,
+    pub cmd_alloc_size_: size_t,
+    pub commands_: *mut Command,
+    pub num_commands_: size_t,
+    pub num_literals_: size_t,
+    pub last_insert_len_: size_t,
+    pub last_flush_pos_: u64,
+    pub last_processed_pos_: u64,
+    pub dist_cache_: [c_int; 4],
+    pub saved_dist_cache_: [c_int; 4],
+    pub last_byte_: u8,
+    pub last_byte_bits_: u8,
+    pub prev_byte_: u8,
+    pub prev_byte2_: u8,
+    pub storage_size_: size_t,
+    pub storage_: *mut u8,
+    pub small_table_: [c_int; 1 << 10],
+    pub large_table_: *mut c_int,
+    pub large_table_size_: size_t,
+    pub cmd_depths_: [u8; 128],
+    pub cmd_bits_: [u16; 128],
+    pub cmd_code_: [u8; 512],
+    pub cmd_code_numbits_: size_t,
+    pub command_buf_: *mut u32,
+    pub literal_buf_: *mut u8,
+    pub next_out_: *mut u8,
+    pub available_out_: size_t,
+    pub total_out_: size_t,
+    pub flush_buf_: [u8; 2],
+    pub stream_state_: BrotliEncoderStreamState,
+    pub is_last_block_emitted_: c_int,
+    pub is_initialized_: c_int,
+}
+
+#[repr(C)]
+pub struct MemoryManager {
+    pub alloc_func: Option<brotli_alloc_func>,
+    pub free_func: Option<brotli_free_func>,
+    pub opaque: *mut c_void,
+}
+
+#[repr(C)]
+pub struct Hashers {
+    pub hash_h2: *mut H2,
+    pub hash_h3: *mut H3,
+    pub hash_h4: *mut H4,
+    pub hash_h5: *mut H5,
+    pub hash_h6: *mut H6,
+    pub hash_h7: *mut H7,
+    pub hash_h8: *mut H8,
+    pub hash_h9: *mut H9,
+    pub hash_h10: *mut H10,
+}
+
+#[repr(C)]
+pub struct H2 {
+    pub buckets_: [u32; (1 << 16) + 1],
```

Comment:
Unfortunately the automated tests for the C bindings I think are choking on this, perhaps the size could be directly inlined here (e.g. calculate `(1 << 16) + 1`)